### DISCUSSION TITLE
Commerce - Strip Out BigCommerce

### DISCUSSION THREAD

**Discussion by fsuaterdogan** [vercel/next.js#22029] [Help]
Hi, we want to develop a prototype commerce project with next.js/commerce. We want to remove BigCommerce in our project. Can anybody help us? Thanks.

**Comment by truecolintracy** [comment]
You'll need to remove the bigcommerce provider under framework and define your types for `Wishlist`, `Customer` and `SearchProductsData`. You can take a look and see that in `framework > commerce > types.ts` there are types that extend types used by bigcommerce. You can also see the Vercel team has a TODO to define those better. But if you're wanting to remove bigcommerce completely you can define the types yourself based off Big Commerce types.

```// TODO: Properly define this type
export type Wishlist = {
  customer?: Customer,
  name: string,
  items: any[]
}

// TODO: Properly define this type
export type Customer = {
  customer?: Customer
}

// TODO: Properly define this type
export type SearchProductsData = {
  products: Product[],
  found: boolean
}
